AddOptionsToCommand: command not found

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Are you using the latest Super-linter version available?

- [X] I am using the latest Super-linter version.
- [X] I can reproduce the issue running Super-linter using complete version identifier (example: vX.Y.Z), and not just with a partial one (example: vX)

- [X] I am using the super-linter/super-linter action or container image, and not the deprecated github/super-linter action or container image.


### Are you resonably sure that it's a Super-linter issue, and not an issue related to a tool that Super-linter runs?

- [X] I think that this is a Super-linter issue.

### Current Behavior

Running super-linter step in Github workflow: 
```
- name: Lint Code Base
  uses: super-linter/super-linter/slim@v7.2.0
  env:
    ...
    KUBERNETES_KUBECONFORM_OPTIONS: -ignore-missing-schemas -schema-location 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' --ignore-filename-pattern '.*tpl\.yaml'
    ...
```
fails with error:
```
HelmRelease ***** failed validation: could not find schema for HelmRelease
```

`HelmRelease` is one of CRDs loaded with `-schema-location` argument passed in `KUBERNETES_KUBECONFORM_OPTIONS`. It seems however, super-linter fails to pass is, since it also reports: 

```
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
```


### Expected Behavior

No error should appear, verified using super-linter `v7.1.0` on exactly the same codebase

### Super-Linter version

```markdown
v7.2.0
```


### Super-linter configuration

```yaml
      - name: Checkout Code
        uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - name: Lint Code Base
        # see https://github.com/marketplace/actions/super-linter#images
        uses: super-linter/super-linter/slim@v7.2.0
        env:
          VALIDATE_ALL_CODEBASE: false
          DEFAULT_BRANCH: master
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

          # Excluding Crunchy Postgres Operator 3rd-party repository from
          # linting. It is fetched as is and has lots of conflicts with our
          # current linter configs.
          FILTER_REGEX_EXCLUDE: ".*/postgres-operator-examples/.*"
          # See https://github.com/yannh/kubeconform#customresourcedefinition-crd-support
          # - Add additional registry with lots of CRDs schemas
          # - If the schema is not present there and is not known - ignore (instead of failure)
          KUBERNETES_KUBECONFORM_OPTIONS: -ignore-missing-schemas -schema-location 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' --ignore-filename-pattern '.*tpl\.yaml'
          # checkov returns too many failed checks at the moment
          VALIDATE_CHECKOV: false
          # Go linter does not work properly with VALIDATE_ALL_CODEBASE false,
          # because it tries to lint files one-by-one and might complain about
          # unused/undefined variables that are used/defined in another file.
          VALIDATE_GO: false
          # It takes DRY to the extreme and annoys, people tend to just ignore it
          VALIDATE_JSCPD: false
          # Disable Markdown prettier until we provide a generic config for developers to run prettier locally
          VALIDATE_MARKDOWN_PRETTIER: false
          # Textlint just discourages documentation
          VALIDATE_NATURAL_LANGUAGE: false
          # Not much Ruby here, validation is up to a developer
          VALIDATE_RUBY: false
          # Disable AWS Step Function states validation
          # We don't use them and current check fails on IAM permissions jsons
          VALIDATE_STATES: false
          # Disable Terrascan
          VALIDATE_TERRAFORM_TERRASCAN: false
          # Disable YAML prettier until we provide a generic config for developers to run prettier locally
          VALIDATE_YAML_PRETTIER: false
          # Disable JSON prettier until we provide a generic config for developers to run prettier locally
          VALIDATE_JSON_PRETTIER: false
```


### Relevant log output

```shell
Run super-linter/super-linter/slim@v7.2.0
  env:
    VALIDATE_ALL_CODEBASE: false
    DEFAULT_BRANCH: master
    GITHUB_TOKEN: ***
    FILTER_REGEX_EXCLUDE: .*/postgres-operator-examples/.*
    KUBERNETES_KUBECONFORM_OPTIONS: -ignore-missing-schemas -schema-location 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' --ignore-filename-pattern '.*tpl\.yaml'
    VALIDATE_CHECKOV: false
    VALIDATE_GO: false
    VALIDATE_JSCPD: false
    VALIDATE_MARKDOWN_PRETTIER: false
    VALIDATE_NATURAL_LANGUAGE: false
    VALIDATE_RUBY: false
    VALIDATE_STATES: false
    VALIDATE_TERRAFORM_TERRASCAN: false
    VALIDATE_YAML_PRETTIER: false
    VALIDATE_JSON_PRETTIER: false
/usr/bin/docker run --name ghcriosuperlintersuperlinterslimv720_b4eba5 --label 0f433a --workdir /github/workspace --rm -e "VALIDATE_ALL_CODEBASE" -e "DEFAULT_BRANCH" -e "GITHUB_TOKEN" -e "FILTER_REGEX_EXCLUDE" -e "KUBERNETES_KUBECONFORM_OPTIONS" -e "VALIDATE_CHECKOV" -e "VALIDATE_GO" -e "VALIDATE_JSCPD" -e "VALIDATE_MARKDOWN_PRETTIER" -e "VALIDATE_NATURAL_LANGUAGE" -e "VALIDATE_RUBY" -e "VALIDATE_STATES" -e "VALIDATE_TERRAFORM_TERRASCAN" -e "VALIDATE_YAML_PRETTIER" -e "VALIDATE_JSON_PRETTIER" -e "HOME" -e "GITHUB_JOB" -e "GITHUB_REF" -e "GITHUB_SHA" -e "GITHUB_REPOSITORY" -e "GITHUB_REPOSITORY_OWNER" -e "GITHUB_REPOSITORY_OWNER_ID" -e "GITHUB_RUN_ID" -e "GITHUB_RUN_NUMBER" -e "GITHUB_RETENTION_DAYS" -e "GITHUB_RUN_ATTEMPT" -e "GITHUB_REPOSITORY_ID" -e "GITHUB_ACTOR_ID" -e "GITHUB_ACTOR" -e "GITHUB_TRIGGERING_ACTOR" -e "GITHUB_WORKFLOW" -e "GITHUB_HEAD_REF" -e "GITHUB_BASE_REF" -e "GITHUB_EVENT_NAME" -e "GITHUB_SERVER_URL" -e "GITHUB_API_URL" -e "GITHUB_GRAPHQL_URL" -e "GITHUB_REF_NAME" -e "GITHUB_REF_PROTECTED" -e "GITHUB_REF_TYPE" -e "GITHUB_WORKFLOW_REF" -e "GITHUB_WORKFLOW_SHA" -e "GITHUB_WORKSPACE" -e "GITHUB_ACTION" -e "GITHUB_EVENT_PATH" -e "GITHUB_ACTION_REPOSITORY" -e "GITHUB_ACTION_REF" -e "GITHUB_PATH" -e "GITHUB_ENV" -e "GITHUB_STEP_SUMMARY" -e "GITHUB_STATE" -e "GITHUB_OUTPUT" -e "RUNNER_OS" -e "RUNNER_ARCH" -e "RUNNER_NAME" -e "RUNNER_ENVIRONMENT" -e "RUNNER_TOOL_CACHE" -e "RUNNER_TEMP" -e "RUNNER_WORKSPACE" -e "ACTIONS_RUNTIME_URL" -e "ACTIONS_RUNTIME_TOKEN" -e "ACTIONS_CACHE_URL" -e "ACTIONS_RESULTS_URL" -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "***":"/github/workspace" ghcr.io/super-linter/super-linter:slim-v7.2.0
Super-Linter initialization
  2024-11-19 11:37:58 [INFO]   --------------------------------------------------------------------------------
  
                                /@@#///////@@/(@//@%/(@.@(       @@
                            @@//////////////////////////////#*  @@@
                          @////@//(///////////@@@@@///@//@/@**//@@(
                        @///////@///////////////@@@@    (           @,
                       @/(&/@////////////////////                     @
                      @////////////////////////@@                      @
                    @%////////(//////////%/////&@            @@       *,@           ______________
               @@@@@/@/#/////(&//////////////////                       .@         /              \
          *@@@@@.    .%///(//@//////////////////&.   .@@,                 @%      / Don't mind me  \
        @@%           .&@&&/@.@//&/////(//////////    @@@@@@@@@         .. &@    / I'm just looking \
      @@%               @@@@@   @&/////////////////#   @/       V  @@/ ,@@@ @   <  for some trash... |
  @@@%                   @@@@        .%@@@@//////#@ @   @@         @     .,.     \__________________/
                                            @@@/@(  (@@@@% @/\      %
                                             @@@@(    .     .@@/\   #
                                               @                  %@%
  
  --------------------------------------------------------------------------------
  2024-11-19 11:37:58 [INFO]   ---------------------------------------------
  2024-11-19 11:37:58 [INFO]   --- GitHub Actions Multi Language Linter ----
  2024-11-19 11:37:58 [INFO]    - Image Creation Date: 2024-11-18T13:53:56Z
  2024-11-19 11:37:58 [INFO]    - Image Revision: e1cb86b6e8d119f789513668b4b30bf17fe1efe4
  2024-11-19 11:37:58 [INFO]    - Image Version: 7.2.0
  2024-11-19 11:37:58 [INFO]   ---------------------------------------------
  2024-11-19 11:37:58 [INFO]   ---------------------------------------------
  2024-11-19 11:37:58 [INFO]   The Super-Linter source code can be found at:
  2024-11-19 11:37:58 [INFO]    - https://github.com/super-linter/super-linter
  2024-11-19 11:37:58 [INFO]   ---------------------------------------------
  2024-11-19 11:37:58 [INFO]   - Validating changed files in code base...
  2024-11-19 11:37:58 [INFO]   This version of Super-linter includes the following tools:
  [ANSIBLE] ansible-lint: 24.10.0
  [BASH] shellcheck: 0.10.0
  [BASH_EXEC] bash-exec: Version command not supported
  [CHECKOV] checkov: 3.2.295
  [CLANG_FORMAT] clang-format: 17.0.6
  [CLOJURE] clj-kondo: v2024.11.14
  [CLOUDFORMATION] cfn-lint: 1.19.0
  [COFFEESCRIPT] coffeelint: 5.2.11
  [CPP] cpplint: 2.0.0
  [CSS] stylelint: 16.10.0
  [CSS_PRETTIER] prettier: 3.3.3
  [DART] dart: 3.5.4
  [DOCKERFILE_HADOLINT] hadolint: 2.12.0
  [EDITORCONFIG] editorconfig-checker: v3.0.3
  [ENV] dotenv-linter: 3.3.0
  [GHERKIN] gherkin-lint: Version command not supported
  [GITHUB_ACTIONS] actionlint: 1.7.4
  [GITLEAKS] gitleaks: v8.21.2
  [GIT_COMMITLINT] commitlint: @commitlint/cli@19.5.0
  [GIT_MERGE_CONFLICT_MARKERS] git-merge-conflict-markers: 1.0.0
  [GOOGLE_JAVA_FORMAT] google-java-format: 1.24.0
  [GO] golangci-lint: 1.62.0
  [GO_MODULES] golangci-lint: 1.62.0
  [GO_RELEASER] goreleaser: 2.4.5
  [GRAPHQL_PRETTIER] prettier: 3.3.3
  [GROOVY] npm-groovy-lint: 15.0.2
  [HTML] htmlhint: 1.1.4
  [HTML_PRETTIER] prettier: 3.3.3
  [JAVASCRIPT_ES] eslint: v8.57.1
  [JAVASCRIPT_PRETTIER] prettier: 3.3.3
  [JAVASCRIPT_STANDARD] standard: 17.1.2
  [JAVA] checkstyle: 10.20.1
  [JSCPD] jscpd: 4.0.5
  [JSONC] eslint: v8.57.1
  [JSONC_PRETTIER] prettier: 3.3.3
  [JSON] eslint: v8.57.1
  [JSON_PRETTIER] prettier: 3.3.3
  [JSX] eslint: v8.57.1
  [JSX_PRETTIER] prettier: 3.3.3
  [KOTLIN] ktlint: 1.4.1
  [KUBERNETES_KUBECONFORM] kubeconform: v0.6.7
  [LATEX] chktex: v1.7.9
  [LUA] lua: 5.4.6
  [MARKDOWN] markdownlint: 0.42.0
  [MARKDOWN_PRETTIER] prettier: 3.3.3
  [NATURAL_LANGUAGE] textlint: v14.3.0
  [OPENAPI] spectral: 6.14.1
  [PERL] perl: (v5.38.2)
  [PHP_BUILTIN] php: 8.3.13
  [PHP_PHPCS] phpcs: 3.11.0
  [PHP_PHPSTAN] phpstan: 2.0.1
  [PHP_PSALM] psalm: 5.26.1@d747f6500b38ac4f7dfc5edbcae6e4b637d7add0
  [PROTOBUF] protolint: 0.50.5(6cc254e)
  [PYTHON_BLACK] black: 24.10.0
  [PYTHON_FLAKE8] flake8: 7.1.1
  [PYTHON_ISORT] isort: 5.13.2
  [PYTHON_MYPY] mypy: 1.13.0
  [PYTHON_PYINK] pyink: 24.10.0
  [PYTHON_PYLINT] pylint: 3.3.1
  [PYTHON_RUFF] ruff: 0.7.3
  [RAKU] raku: v2024.04
  [RENOVATE] renovate-config-validator: 39.15.4
  [RUBY] rubocop: 1.68.0
  [R] R: 4.4.0
  [SCALAFMT] scalafmt: 3.8.3
  [SHELL_SHFMT] shfmt: v3.10.0
  [SNAKEMAKE_LINT] snakemake: 8.25.3
  [SNAKEMAKE_SNAKEFMT] snakefmt: 0.10.2
  [SQLFLUFF] sqlfluff: 3.2.5
  [STATES] asl-validator: 3.9.0
  [TEKTON] tekton-lint: 1.1.0
  [TERRAFORM_FMT] terraform: v1.9.8
  [TERRAFORM_TERRASCAN] terrascan: v1.19.9
  [TERRAFORM_TFLINT] tflint: 0.54.0
  [TERRAGRUNT] terragrunt: version
  [TSX] eslint: v8.57.1
  [TYPESCRIPT_ES] eslint: v8.57.1
  [TYPESCRIPT_PRETTIER] prettier: 3.3.3
  [TYPESCRIPT_STANDARD] ts-standard: 12.0.2
  [VUE_PRETTIER] prettier: 3.3.3
  [XML] xmllint: 21207
  [YAML] yamllint: 1.35.1
  [YAML_PRETTIER] prettier: 3.3.3
  2024-11-19 11:37:58 [INFO]   --------------------------------------------
  2024-11-19 11:37:58 [INFO]   Gathering GitHub information...
  2024-11-19 11:37:58 [INFO]   Successfully validated GITHUB_WORKSPACE: /github/workspace
  2024-11-19 11:37:58 [INFO]   Successfully found GITHUB_EVENT_PATH: /github/workflow/event.json]
  2024-11-19 11:37:58 [INFO]   Successfully found GITHUB_SHA: ***
  2024-11-19 11:37:59 [INFO]   Successfully found GITHUB_ORG: ***
  2024-11-19 11:37:59 [INFO]   Successfully found GITHUB_REPO: ***
  2024-11-19 11:37:59 [INFO]   The default branch for this repository is set to: ***
  2024-11-19 11:37:59 [INFO]   Successfully found GITHUB_TOKEN.
  2024-11-19 11:37:59 [INFO]   Successfully found GITHUB_REPOSITORY: ***
  2024-11-19 11:37:59 [INFO]   Successfully found GITHUB_RUN_ID ***
  2024-11-19 11:37:59 [INFO]   --------------------------------------------
  2024-11-19 11:37:59 [INFO]   Validating the configuration
  .git
  2024-11-19 11:38:00 [INFO]   The master branch doesn't exist in /github/workspace
  ***
  2024-11-19 11:38:00 [INFO]   *** doesn't exist, however origin/*** exists. Setting DEFAULT_BRANCH to: origin/***
  2024-11-19 11:38:00 [WARN]   Git commit message validation with commitlint is enabled, but no commitlint configuration file is available. Disabling commitlint. To suppress this message, either disable Git commit validation by setting VALIDATE_GIT_COMMITLINT to false in your Super-linter configuration, or provide a commitlint configuration file.
  2024-11-19 11:38:01 [INFO]   ----------------------------------------------
  2024-11-19 11:38:01 [INFO]   User provided file:[/github/workspace/.github/linters/actionlint.yml] exists, setting rules file...
  2024-11-19 11:38:01 [INFO]   ----------------------------------------------
  2024-11-19 11:38:01 [INFO]   User provided file:[/github/workspace/.github/linters/.gitleaks.toml] exists, setting rules file...
  2024-11-19 11:38:01 [INFO]   ----------------------------------------------
  2024-11-19 11:38:01 [INFO]   User provided file:[/github/workspace/.github/linters/.golangci.yml] exists, setting rules file...
  2024-11-19 11:38:01 [INFO]   ----------------------------------------------
  2024-11-19 11:38:01 [INFO]   User provided file:[/github/workspace/.github/linters/.textlintrc] exists, setting rules file...
  2024-11-19 11:38:02 [INFO]   ----------------------------------------------
  2024-11-19 11:38:02 [INFO]   User provided file:[/github/workspace/.github/linters/.tflint.hcl] exists, setting rules file...
  2024-11-19 11:38:02 [INFO]   ----------------------------------------------
  2024-11-19 11:38:02 [INFO]   User provided file:[/github/workspace/.github/linters/.yaml-lint.yml] exists, setting rules file...
  2024-11-19 11:38:04 [INFO]   Building the list of files and directories to check
  2024-11-19 11:38:04 [INFO]   Successfully gathered list of files...
2024-11-19 11:38:07 [INFO]   Command output when running linters:
------
EDITORCONFIG
  2024-11-19 11:38:05 [INFO]   Linting EDITORCONFIG items...
  Notice: 19 11:38:05 [NOTICE]   EDITORCONFIG linted successfully
GITLEAKS
  2024-11-19 11:38:05 [INFO]   Linting GITLEAKS items...
  Notice: 19 11:38:05 [NOTICE]   GITLEAKS linted successfully
  2024-11-19 11:38:05 [INFO]   Stderr contents for GITLEAKS:
  ------
  11:38AM INF scan completed in 225µs
  11:38AM INF no leaks found
  ------
GIT_MERGE_CONFLICT_MARKERS
  2024-11-19 11:38:05 [INFO]   Linting GIT_MERGE_CONFLICT_MARKERS items...
  Notice: 19 11:38:05 [NOTICE]   GIT_MERGE_CONFLICT_MARKERS linted successfully
  2024-11-19 11:38:05 [INFO]   Command output for GIT_MERGE_CONFLICT_MARKERS:
  ------
  No merge conflicts found in /github/workspace/***/helm-release.yaml
  ------
KUBERNETES_KUBECONFORM
  2024-11-19 11:38:06 [INFO]   Linting KUBERNETES_KUBECONFORM items...
  Error: -19 11:38:06 [ERROR]   Found errors when linting KUBERNETES_KUBECONFORM. Exit code: 1.
  2024-11-19 11:38:06 [INFO]   Command output for KUBERNETES_KUBECONFORM:
  ------
  /github/workspace/***/helm-release.yaml - HelmRelease *** failed validation: could not find schema for HelmRelease
  ------
YAML
  2024-11-19 11:38:07 [INFO]   Linting YAML items...
  Notice: 19 11:38:07 [NOTICE]   YAML linted successfully
  2024-11-19 11:38:07 [INFO]   Command output for YAML:
  ------
  Warning: orkspace/***/helm-release.yaml:1:1: [warning] missing document start "---" (document-start)
  Warning: orkspace/***/helm-release.yaml:18:81: [warning] line too long (87 > 80 characters) (line-length)
  Warning: orkspace/***/helm-release.yaml:28:81: [warning] line too long (104 > 80 characters) (line-length)
  Warning: orkspace/***/helm-release.yaml:30:81: [warning] line too long (104 > 80 characters) (line-length)
  ------
------
2024-11-19 11:38:07 [INFO]   Stderr when running linters:
------
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
/action/lib/functions/linterCommands.sh: line 111: AddOptionsToCommand: command not found
------
2024-11-19 11:38:07 [INFO]   ----------------------------------------------
2024-11-19 11:38:07 [INFO]   ----------------------------------------------
Notice: 19 11:38:07 [NOTICE]   Successfully linted EDITORCONFIG
Notice: 19 11:38:08 [NOTICE]   Successfully linted GITLEAKS
Notice: 19 11:38:09 [NOTICE]   Successfully linted GIT_MERGE_CONFLICT_MARKERS
Error: -19 11:38:09 [ERROR]   Errors found in KUBERNETES_KUBECONFORM
Notice: 19 11:38:10 [NOTICE]   Successfully linted YAML
Error: -19 11:38:10 [ERROR]   Super-linter detected linting errors
super-linter-output/super-linter-summary.md 43ms
```


### Steps To Reproduce

n/a

### Anything else?

_No response_

--- a/Makefile
+++ b/Makefile
@@ -4,7 +4,7 @@
 all: info docker test ## Run all targets.
 
 .PHONY: test
-test: info validate-container-image-labels docker-build-check docker-dev-container-build-check npm-audit test-lib inspec lint-codebase fix-codebase test-default-config-files test-actions-runner-debug test-actions-steps-debug test-runner-debug test-find lint-subset-files test-custom-ssl-cert test-non-default-workdir test-git-flags test-non-default-home-directory test-git-initial-commit test-git-merge-commit-push test-log-level test-use-find-and-ignore-gitignored-files test-linters-expect-failure-log-level-notice test-bash-exec-library-expect-success test-bash-exec-library-expect-failure test-save-super-linter-output test-save-super-linter-output-custom-path test-save-super-linter-custom-summary test-custom-gitleaks-log-level test-dont-save-super-linter-log-file test-dont-save-super-linter-output test-linters test-linters-fix-mode ## Run the test suite
+test: info validate-container-image-labels docker-build-check docker-dev-container-build-check npm-audit test-lib inspec lint-codebase fix-codebase test-default-config-files test-actions-runner-debug test-actions-steps-debug test-runner-debug test-find lint-subset-files test-custom-ssl-cert test-non-default-workdir test-git-flags test-non-default-home-directory test-git-initial-commit test-git-merge-commit-push test-log-level test-use-find-and-ignore-gitignored-files test-linters-expect-failure-log-level-notice test-bash-exec-library-expect-success test-bash-exec-library-expect-failure test-save-super-linter-output test-save-super-linter-output-custom-path test-save-super-linter-custom-summary test-custom-gitleaks-log-level test-dont-save-super-linter-log-file test-dont-save-super-linter-output test-linter-command-options test-linters test-linters-fix-mode ## Run the test suite
 
 # if this session isn't interactive, then we don't want to allocate a
 # TTY, which would fail, but if it is interactive, we do want to attach
@@ -606,6 +606,13 @@ test-dont-save-super-linter-output: ## Run super-linter without saving Super-lin
 	$(CURDIR)/test/run-super-linter-tests.sh \
 		$(SUPER_LINTER_TEST_CONTAINER_URL) \
 		"run_test_case_dont_save_super_linter_output" \
+		"$(IMAGE)"
+
+.PHONY: test-linter-command-options
+test-linter-command-options: ## Run super-linter passing options to linters
+	$(CURDIR)/test/run-super-linter-tests.sh \
+		$(SUPER_LINTER_TEST_CONTAINER_URL) \
+		"run_test_case_linter_command_options" \
 		"$(IMAGE)"
 
 .PHONY: docker-dev-container-build-check ## Run Docker build checks against the dev-container image

--- a/lib/functions/linterCommands.sh
+++ b/lib/functions/linterCommands.sh
@@ -8,178 +8,7 @@
 # shellcheck source=/dev/null
 source /action/lib/globals/linterCommandsOptions.sh
 
-##########################
-# Define linter commands #
-##########################
-
-# These commands are reused across several languages
-PRETTIER_COMMAND=(prettier)
-DOTNET_FORMAT_COMMAND=(dotnet format)
-
-LINTER_COMMANDS_ARRAY_ANSIBLE=(ansible-lint -c "${ANSIBLE_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_ARM=(pwsh -NoProfile -NoLogo -Command "\"Import-Module ${ARM_TTK_PSD1} ; \\\${config} = \\\$(Import-PowerShellDataFile -Path ${ARM_LINTER_RULES}) ; Test-AzTemplate @config -TemplatePath '{}'; if (\\\${Error}.Count) { exit 1 }\"")
-LINTER_COMMANDS_ARRAY_BASH=(shellcheck --color --rcfile "${BASH_LINTER_RULES}")
-# This check and the BASH_SEVERITY variable are needed until Shellcheck supports
-# setting severity using its config file.
-# Ref: https://github.com/koalaman/shellcheck/issues/2178
-if [ -n "${BASH_SEVERITY}" ]; then
-  export BASH_SEVERITY
-  LINTER_COMMANDS_ARRAY_BASH+=(--severity="${BASH_SEVERITY}")
-fi
-LINTER_COMMANDS_ARRAY_BASH_EXEC=(bash-exec '{}')
-if [ "${BASH_EXEC_IGNORE_LIBRARIES}" == 'true' ]; then
-  debug "Enabling bash-exec option to ignore shell library files."
-  LINTER_COMMANDS_ARRAY_BASH_EXEC+=('true')
-fi
-LINTER_COMMANDS_ARRAY_CHECKOV=(checkov --config-file "${CHECKOV_LINTER_RULES}")
-if CheckovConfigurationFileContainsDirectoryOption "${CHECKOV_LINTER_RULES}"; then
-  # Consume the input as we do with ANSIBLE
-  debug "Consume the input of the Checkov command because we don't need to add it as an argument."
-  LINTER_COMMANDS_ARRAY_CHECKOV+=("&& echo \"Got the list of directories to lint from the configuration file: {}\"")
-else
-  debug "Adding the '--directory' option to the Checkov command."
-  LINTER_COMMANDS_ARRAY_CHECKOV+=(--directory)
-fi
-LINTER_COMMANDS_ARRAY_CLANG_FORMAT=(clang-format --style=file:"${CLANG_FORMAT_LINTER_RULES}" --Werror)
-LINTER_COMMANDS_ARRAY_CLOJURE=(clj-kondo --config "${CLOJURE_LINTER_RULES}" --lint)
-LINTER_COMMANDS_ARRAY_CLOUDFORMATION=(cfn-lint --config-file "${CLOUDFORMATION_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_COFFEESCRIPT=(coffeelint -f "${COFFEESCRIPT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_CPP=(cpplint)
-LINTER_COMMANDS_ARRAY_CSHARP=("${DOTNET_FORMAT_COMMAND[@]}" whitespace --folder --exclude / --include "{/}")
-LINTER_COMMANDS_ARRAY_CSS=(stylelint --config "${CSS_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_CSS_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_DART=(dart analyze --fatal-infos --fatal-warnings)
-LINTER_COMMANDS_ARRAY_DOCKERFILE_HADOLINT=(hadolint -c "${DOCKERFILE_HADOLINT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_ANALYZERS=("${DOTNET_FORMAT_COMMAND[@]}" analyzers)
-LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_STYLE=("${DOTNET_FORMAT_COMMAND[@]}" style)
-LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_WHITESPACE=("${DOTNET_FORMAT_COMMAND[@]}" whitespace)
-LINTER_COMMANDS_ARRAY_EDITORCONFIG=(editorconfig-checker -config "${EDITORCONFIG_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_ENV=(dotenv-linter)
-LINTER_COMMANDS_ARRAY_GITHUB_ACTIONS=(actionlint -config-file "${GITHUB_ACTIONS_LINTER_RULES}")
-if [ "${GITHUB_ACTIONS_COMMAND_ARGS}" != "null" ] && [ -n "${GITHUB_ACTIONS_COMMAND_ARGS}" ]; then
-  export GITHUB_ACTIONS_COMMAND_ARGS
-  AddOptionsToCommand "LINTER_COMMANDS_ARRAY_GITHUB_ACTIONS" "${GITHUB_ACTIONS_COMMAND_ARGS}"
-fi
-LINTER_COMMANDS_ARRAY_GITLEAKS=(gitleaks detect --no-banner --no-git --redact --config "${GITLEAKS_LINTER_RULES}" --verbose)
-if [ -n "${GITLEAKS_LOG_LEVEL:-}" ]; then
-  export GITLEAKS_LOG_LEVEL
-  LINTER_COMMANDS_ARRAY_GITLEAKS+=("${GITLEAKS_LOG_LEVEL_OPTIONS[@]}" "${GITLEAKS_LOG_LEVEL}")
-  debug "Add log options to the Gitleaks command: ${LINTER_COMMANDS_ARRAY_GITLEAKS[*]}"
-fi
-LINTER_COMMANDS_ARRAY_GITLEAKS+=(--source)
-LINTER_COMMANDS_ARRAY_GHERKIN=(gherkin-lint -c "${GHERKIN_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_GIT_COMMITLINT=(commitlint --verbose --cwd "{}")
-if [[ -n "${GITHUB_BEFORE_SHA:-}" ]]; then
-  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=(--from "${GITHUB_BEFORE_SHA}" --to "${GITHUB_SHA}")
-else
-  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=(--last)
-fi
-if [ "${ENABLE_COMMITLINT_STRICT_MODE:-}" == 'true' ]; then
-  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=("${COMMITLINT_STRICT_MODE_OPTIONS[@]}")
-fi
-LINTER_COMMANDS_ARRAY_GIT_MERGE_CONFLICT_MARKERS=(git-merge-conflict-markers)
-LINTER_COMMANDS_ARRAY_GO=(golangci-lint run -c "${GO_LINTER_RULES}" --fast)
-LINTER_COMMANDS_ARRAY_GO_MODULES=(golangci-lint run --allow-parallel-runners -c "${GO_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_GO_RELEASER=(goreleaser check)
-LINTER_COMMANDS_ARRAY_GOOGLE_JAVA_FORMAT=(java -jar /usr/bin/google-java-format)
-LINTER_COMMANDS_ARRAY_GRAPHQL_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_GROOVY=(npm-groovy-lint -c "${GROOVY_LINTER_RULES}" --failon warning --no-insight)
-LINTER_COMMANDS_ARRAY_HTML=(htmlhint --config "${HTML_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_HTML_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_JAVA=(java -jar /usr/bin/checkstyle -c "${JAVA_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_JAVASCRIPT_ES=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_JAVASCRIPT_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_JAVASCRIPT_STANDARD=(standard "${JAVASCRIPT_STANDARD_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_JSCPD=(jscpd --config "${JSCPD_LINTER_RULES}")
-JSCPD_GITIGNORE_OPTION="--gitignore"
-if [[ "${IGNORE_GITIGNORED_FILES}" == "true" ]]; then
-  debug "IGNORE_GITIGNORED_FILES is ${IGNORE_GITIGNORED_FILES}. Enable Jscpd option to ignore files that Git ignores (${JSCPD_GITIGNORE_OPTION})"
-  # Users can also add the '"gitignore": true' option in their Jscpd config files to achieve the same functionality
-  # but we want to respect IGNORE_GITIGNORED_FILES
-  LINTER_COMMANDS_ARRAY_JSCPD+=("${JSCPD_GITIGNORE_OPTION}")
-fi
-LINTER_COMMANDS_ARRAY_JSON=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}" --ext '.json')
-LINTER_COMMANDS_ARRAY_JSON_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_JSONC=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}" --ext '.json5,.jsonc')
-LINTER_COMMANDS_ARRAY_JSONC_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_JSX=(eslint -c "${JSX_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_JSX_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_KOTLIN=(ktlint "{/}")
-LINTER_COMMANDS_ARRAY_KUBERNETES_KUBECONFORM=(kubeconform -strict)
-if [ "${KUBERNETES_KUBECONFORM_OPTIONS}" != "null" ] && [ -n "${KUBERNETES_KUBECONFORM_OPTIONS}" ]; then
-  export KUBERNETES_KUBECONFORM_OPTIONS
-  AddOptionsToCommand "LINTER_COMMANDS_ARRAY_KUBERNETES_KUBECONFORM" "${KUBERNETES_KUBECONFORM_OPTIONS}"
-fi
-LINTER_COMMANDS_ARRAY_LATEX=(chktex -q -l "${LATEX_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_LUA=(luacheck --config "${LUA_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_MARKDOWN=(markdownlint -c "${MARKDOWN_LINTER_RULES}")
-if [ -n "${MARKDOWN_CUSTOM_RULE_GLOBS}" ]; then
-  IFS="," read -r -a MARKDOWN_CUSTOM_RULE_GLOBS_ARRAY <<<"${MARKDOWN_CUSTOM_RULE_GLOBS}"
-  for glob in "${MARKDOWN_CUSTOM_RULE_GLOBS_ARRAY[@]}"; do
-    if [ -z "${LINTER_RULES_PATH}" ]; then
-      LINTER_COMMANDS_ARRAY_MARKDOWN+=(-r "${GITHUB_WORKSPACE}/${glob}")
-    else
-      LINTER_COMMANDS_ARRAY_MARKDOWN+=(-r "${GITHUB_WORKSPACE}/${LINTER_RULES_PATH}/${glob}")
-    fi
-  done
-fi
-LINTER_COMMANDS_ARRAY_MARKDOWN_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_NATURAL_LANGUAGE=(textlint -c "${NATURAL_LANGUAGE_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_OPENAPI=(spectral lint -r "${OPENAPI_LINTER_RULES}" -D)
-LINTER_COMMANDS_ARRAY_PERL=(perlcritic)
-if [ "${PERL_PERLCRITIC_OPTIONS}" != "null" ] && [ -n "${PERL_PERLCRITIC_OPTIONS}" ]; then
-  export PERL_PERLCRITIC_OPTIONS
-  AddOptionsToCommand "LINTER_COMMANDS_ARRAY_PERL" "${PERL_PERLCRITIC_OPTIONS}"
-fi
-LINTER_COMMANDS_ARRAY_PHP_BUILTIN=(php -l -c "${PHP_BUILTIN_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PHP_PHPCS=(phpcs --standard="${PHP_PHPCS_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PHP_PHPSTAN=(phpstan analyse --no-progress --no-ansi --memory-limit 1G -c "${PHP_PHPSTAN_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PHP_PSALM=(psalm --config="${PHP_PSALM_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_POWERSHELL=(Invoke-ScriptAnalyzer -EnableExit -Settings "${POWERSHELL_LINTER_RULES}" -Path '{}')
-LINTER_COMMANDS_ARRAY_PROTOBUF=(protolint lint -config_path "${PROTOBUF_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_BLACK=(black --config "${PYTHON_BLACK_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_PYLINT=(pylint --rcfile "${PYTHON_PYLINT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_FLAKE8=(flake8 --config="${PYTHON_FLAKE8_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_ISORT=(isort --sp "${PYTHON_ISORT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_MYPY=(mypy --config-file "${PYTHON_MYPY_LINTER_RULES}" --install-types --non-interactive)
-LINTER_COMMANDS_ARRAY_PYTHON_PYINK=(pyink --config "${PYTHON_PYINK_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_PYTHON_RUFF=(ruff check --config "${PYTHON_RUFF_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_R=(R --slave -e "\"lints <- lintr::lint('{}');print(lints);errors <- purrr::keep(lints, ~ .\\\$type == 'error');quit(save = 'no', status = if (length(errors) > 0) 1 else 0)\"")
-LINTER_COMMANDS_ARRAY_RAKU=(raku)
-LINTER_COMMANDS_ARRAY_RENOVATE=(renovate-config-validator --strict)
-LINTER_COMMANDS_ARRAY_RUBY=(rubocop -c "${RUBY_LINTER_RULES}" --force-exclusion --ignore-unrecognized-cops)
-LINTER_COMMANDS_ARRAY_RUST_2015=(rustfmt --edition 2015)
-LINTER_COMMANDS_ARRAY_RUST_2018=(rustfmt --edition 2018)
-LINTER_COMMANDS_ARRAY_RUST_2021=(rustfmt --edition 2021)
-LINTER_COMMANDS_ARRAY_RUST_CLIPPY=(cargo clippy)
-if [ -n "${RUST_CLIPPY_COMMAND_OPTIONS:-}" ]; then
-  export RUST_CLIPPY_COMMAND_OPTIONS
-  AddOptionsToCommand "LINTER_COMMANDS_ARRAY_RUST_CLIPPY" "${RUST_CLIPPY_COMMAND_OPTIONS}"
-fi
-LINTER_COMMANDS_ARRAY_SCALAFMT=(scalafmt --config "${SCALAFMT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_SHELL_SHFMT=(shfmt)
-LINTER_COMMANDS_ARRAY_SNAKEMAKE_LINT=(snakemake --lint -s)
-LINTER_COMMANDS_ARRAY_SNAKEMAKE_SNAKEFMT=(snakefmt --config "${SNAKEMAKE_SNAKEFMT_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_STATES=(asl-validator --json-path)
-LINTER_COMMANDS_ARRAY_SQLFLUFF=(sqlfluff)
-LINTER_COMMANDS_ARRAY_TEKTON=(tekton-lint)
-LINTER_COMMANDS_ARRAY_TERRAFORM_FMT=(terraform fmt)
-LINTER_COMMANDS_ARRAY_TERRAFORM_TFLINT=("TF_DATA_DIR=\"/tmp/.terraform-TERRAFORM_TFLINT-{//}\"" tflint -c "${TERRAFORM_TFLINT_LINTER_RULES}" "--filter=\"{/}\"")
-LINTER_COMMANDS_ARRAY_TERRAFORM_TERRASCAN=(terrascan scan -i terraform -t all -c "${TERRAFORM_TERRASCAN_LINTER_RULES}" -f)
-LINTER_COMMANDS_ARRAY_TERRAGRUNT=(terragrunt hclfmt --terragrunt-check --terragrunt-log-level error --terragrunt-hclfmt-file)
-LINTER_COMMANDS_ARRAY_TSX=(eslint -c "${TSX_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_TYPESCRIPT_ES=(eslint -c "${TYPESCRIPT_ES_LINTER_RULES}")
-LINTER_COMMANDS_ARRAY_TYPESCRIPT_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_TYPESCRIPT_STANDARD=(ts-standard --parser @typescript-eslint/parser --plugin @typescript-eslint/eslint-plugin --project "${TYPESCRIPT_STANDARD_TSCONFIG_FILE}")
-LINTER_COMMANDS_ARRAY_VUE_PRETTIER=("${PRETTIER_COMMAND[@]}")
-LINTER_COMMANDS_ARRAY_XML=(xmllint)
-LINTER_COMMANDS_ARRAY_YAML=(yamllint -c "${YAML_LINTER_RULES}" -f parsable)
-if [ "${YAML_ERROR_ON_WARNING}" == 'true' ]; then
-  LINTER_COMMANDS_ARRAY_YAML+=(--strict)
-fi
-LINTER_COMMANDS_ARRAY_YAML_PRETTIER=("${PRETTIER_COMMAND[@]}")
-
-function InitFixModeOptionsAndCommands() {
+InitFixModeOptionsAndCommands() {
   local LANGUAGE="${1}"
   local FIX_MODE_VARIABLE_NAME="FIX_${LANGUAGE}"
   debug "Check if ${LANGUAGE} command needs check only mode or fix mode options or commands by checking if ${FIX_MODE_VARIABLE_NAME} variable is defined."
@@ -259,3 +88,182 @@ AddOptionsToCommand() {
 
   unset -n COMMAND_ARRAY_NAME
 }
+
+##########################
+# Define linter commands #
+##########################
+
+# These commands are reused across several languages
+PRETTIER_COMMAND=(prettier)
+DOTNET_FORMAT_COMMAND=(dotnet format)
+
+LINTER_COMMANDS_ARRAY_ANSIBLE=(ansible-lint -c "${ANSIBLE_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_ARM=(pwsh -NoProfile -NoLogo -Command "\"Import-Module ${ARM_TTK_PSD1} ; \\\${config} = \\\$(Import-PowerShellDataFile -Path ${ARM_LINTER_RULES}) ; Test-AzTemplate @config -TemplatePath '{}'; if (\\\${Error}.Count) { exit 1 }\"")
+LINTER_COMMANDS_ARRAY_BASH=(shellcheck --color --rcfile "${BASH_LINTER_RULES}")
+# This check and the BASH_SEVERITY variable are needed until Shellcheck supports
+# setting severity using its config file.
+# Ref: https://github.com/koalaman/shellcheck/issues/2178
+if [ -n "${BASH_SEVERITY}" ]; then
+  export BASH_SEVERITY
+  LINTER_COMMANDS_ARRAY_BASH+=(--severity="${BASH_SEVERITY}")
+fi
+LINTER_COMMANDS_ARRAY_BASH_EXEC=(bash-exec '{}')
+if [ "${BASH_EXEC_IGNORE_LIBRARIES}" == 'true' ]; then
+  debug "Enabling bash-exec option to ignore shell library files."
+  LINTER_COMMANDS_ARRAY_BASH_EXEC+=('true')
+fi
+LINTER_COMMANDS_ARRAY_CHECKOV=(checkov --config-file "${CHECKOV_LINTER_RULES}")
+if CheckovConfigurationFileContainsDirectoryOption "${CHECKOV_LINTER_RULES}"; then
+  # Consume the input as we do with ANSIBLE
+  debug "Consume the input of the Checkov command because we don't need to add it as an argument."
+  LINTER_COMMANDS_ARRAY_CHECKOV+=("&& echo \"Got the list of directories to lint from the configuration file: {}\"")
+else
+  debug "Adding the '--directory' option to the Checkov command."
+  LINTER_COMMANDS_ARRAY_CHECKOV+=(--directory)
+fi
+LINTER_COMMANDS_ARRAY_CLANG_FORMAT=(clang-format --style=file:"${CLANG_FORMAT_LINTER_RULES}" --Werror)
+LINTER_COMMANDS_ARRAY_CLOJURE=(clj-kondo --config "${CLOJURE_LINTER_RULES}" --lint)
+LINTER_COMMANDS_ARRAY_CLOUDFORMATION=(cfn-lint --config-file "${CLOUDFORMATION_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_COFFEESCRIPT=(coffeelint -f "${COFFEESCRIPT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_CPP=(cpplint)
+LINTER_COMMANDS_ARRAY_CSHARP=("${DOTNET_FORMAT_COMMAND[@]}" whitespace --folder --exclude / --include "{/}")
+LINTER_COMMANDS_ARRAY_CSS=(stylelint --config "${CSS_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_CSS_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_DART=(dart analyze --fatal-infos --fatal-warnings)
+LINTER_COMMANDS_ARRAY_DOCKERFILE_HADOLINT=(hadolint -c "${DOCKERFILE_HADOLINT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_ANALYZERS=("${DOTNET_FORMAT_COMMAND[@]}" analyzers)
+LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_STYLE=("${DOTNET_FORMAT_COMMAND[@]}" style)
+LINTER_COMMANDS_ARRAY_DOTNET_SLN_FORMAT_WHITESPACE=("${DOTNET_FORMAT_COMMAND[@]}" whitespace)
+LINTER_COMMANDS_ARRAY_EDITORCONFIG=(editorconfig-checker -config "${EDITORCONFIG_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_ENV=(dotenv-linter)
+LINTER_COMMANDS_ARRAY_GITHUB_ACTIONS=(actionlint -config-file "${GITHUB_ACTIONS_LINTER_RULES}")
+if [ "${GITHUB_ACTIONS_COMMAND_ARGS}" != "null" ] && [ -n "${GITHUB_ACTIONS_COMMAND_ARGS}" ]; then
+  export GITHUB_ACTIONS_COMMAND_ARGS
+  if ! AddOptionsToCommand "LINTER_COMMANDS_ARRAY_GITHUB_ACTIONS" "${GITHUB_ACTIONS_COMMAND_ARGS}"; then
+    fatal "Error while adding options to GitHub Actions command"
+  fi
+fi
+LINTER_COMMANDS_ARRAY_GITLEAKS=(gitleaks detect --no-banner --no-git --redact --config "${GITLEAKS_LINTER_RULES}" --verbose)
+if [ -n "${GITLEAKS_LOG_LEVEL:-}" ]; then
+  export GITLEAKS_LOG_LEVEL
+  LINTER_COMMANDS_ARRAY_GITLEAKS+=("${GITLEAKS_LOG_LEVEL_OPTIONS[@]}" "${GITLEAKS_LOG_LEVEL}")
+  debug "Add log options to the Gitleaks command: ${LINTER_COMMANDS_ARRAY_GITLEAKS[*]}"
+fi
+LINTER_COMMANDS_ARRAY_GITLEAKS+=(--source)
+LINTER_COMMANDS_ARRAY_GHERKIN=(gherkin-lint -c "${GHERKIN_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_GIT_COMMITLINT=(commitlint --verbose --cwd "{}")
+if [[ -n "${GITHUB_BEFORE_SHA:-}" ]]; then
+  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=(--from "${GITHUB_BEFORE_SHA}" --to "${GITHUB_SHA}")
+else
+  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=(--last)
+fi
+if [ "${ENABLE_COMMITLINT_STRICT_MODE:-}" == 'true' ]; then
+  LINTER_COMMANDS_ARRAY_GIT_COMMITLINT+=("${COMMITLINT_STRICT_MODE_OPTIONS[@]}")
+fi
+LINTER_COMMANDS_ARRAY_GIT_MERGE_CONFLICT_MARKERS=(git-merge-conflict-markers)
+LINTER_COMMANDS_ARRAY_GO=(golangci-lint run -c "${GO_LINTER_RULES}" --fast)
+LINTER_COMMANDS_ARRAY_GO_MODULES=(golangci-lint run --allow-parallel-runners -c "${GO_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_GO_RELEASER=(goreleaser check)
+LINTER_COMMANDS_ARRAY_GOOGLE_JAVA_FORMAT=(java -jar /usr/bin/google-java-format)
+LINTER_COMMANDS_ARRAY_GRAPHQL_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_GROOVY=(npm-groovy-lint -c "${GROOVY_LINTER_RULES}" --failon warning --no-insight)
+LINTER_COMMANDS_ARRAY_HTML=(htmlhint --config "${HTML_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_HTML_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_JAVA=(java -jar /usr/bin/checkstyle -c "${JAVA_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_JAVASCRIPT_ES=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_JAVASCRIPT_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_JAVASCRIPT_STANDARD=(standard "${JAVASCRIPT_STANDARD_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_JSCPD=(jscpd --config "${JSCPD_LINTER_RULES}")
+JSCPD_GITIGNORE_OPTION="--gitignore"
+if [[ "${IGNORE_GITIGNORED_FILES}" == "true" ]]; then
+  debug "IGNORE_GITIGNORED_FILES is ${IGNORE_GITIGNORED_FILES}. Enable Jscpd option to ignore files that Git ignores (${JSCPD_GITIGNORE_OPTION})"
+  # Users can also add the '"gitignore": true' option in their Jscpd config files to achieve the same functionality
+  # but we want to respect IGNORE_GITIGNORED_FILES
+  LINTER_COMMANDS_ARRAY_JSCPD+=("${JSCPD_GITIGNORE_OPTION}")
+fi
+LINTER_COMMANDS_ARRAY_JSON=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}" --ext '.json')
+LINTER_COMMANDS_ARRAY_JSON_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_JSONC=(eslint -c "${JAVASCRIPT_ES_LINTER_RULES}" --ext '.json5,.jsonc')
+LINTER_COMMANDS_ARRAY_JSONC_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_JSX=(eslint -c "${JSX_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_JSX_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_KOTLIN=(ktlint "{/}")
+LINTER_COMMANDS_ARRAY_KUBERNETES_KUBECONFORM=(kubeconform -strict)
+if [ "${KUBERNETES_KUBECONFORM_OPTIONS}" != "null" ] && [ -n "${KUBERNETES_KUBECONFORM_OPTIONS}" ]; then
+  export KUBERNETES_KUBECONFORM_OPTIONS
+  if ! AddOptionsToCommand "LINTER_COMMANDS_ARRAY_KUBERNETES_KUBECONFORM" "${KUBERNETES_KUBECONFORM_OPTIONS}"; then
+    fatal "Error while adding options to Kubeconform command"
+  fi
+fi
+LINTER_COMMANDS_ARRAY_LATEX=(chktex -q -l "${LATEX_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_LUA=(luacheck --config "${LUA_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_MARKDOWN=(markdownlint -c "${MARKDOWN_LINTER_RULES}")
+if [ -n "${MARKDOWN_CUSTOM_RULE_GLOBS}" ]; then
+  IFS="," read -r -a MARKDOWN_CUSTOM_RULE_GLOBS_ARRAY <<<"${MARKDOWN_CUSTOM_RULE_GLOBS}"
+  for glob in "${MARKDOWN_CUSTOM_RULE_GLOBS_ARRAY[@]}"; do
+    if [ -z "${LINTER_RULES_PATH}" ]; then
+      LINTER_COMMANDS_ARRAY_MARKDOWN+=(-r "${GITHUB_WORKSPACE}/${glob}")
+    else
+      LINTER_COMMANDS_ARRAY_MARKDOWN+=(-r "${GITHUB_WORKSPACE}/${LINTER_RULES_PATH}/${glob}")
+    fi
+  done
+fi
+LINTER_COMMANDS_ARRAY_MARKDOWN_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_NATURAL_LANGUAGE=(textlint -c "${NATURAL_LANGUAGE_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_OPENAPI=(spectral lint -r "${OPENAPI_LINTER_RULES}" -D)
+LINTER_COMMANDS_ARRAY_PERL=(perlcritic)
+if [ "${PERL_PERLCRITIC_OPTIONS}" != "null" ] && [ -n "${PERL_PERLCRITIC_OPTIONS}" ]; then
+  export PERL_PERLCRITIC_OPTIONS
+  if ! AddOptionsToCommand "LINTER_COMMANDS_ARRAY_PERL" "${PERL_PERLCRITIC_OPTIONS}"; then
+    fatal "Error while adding options to Perlcritic command"
+  fi
+fi
+LINTER_COMMANDS_ARRAY_PHP_BUILTIN=(php -l -c "${PHP_BUILTIN_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PHP_PHPCS=(phpcs --standard="${PHP_PHPCS_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PHP_PHPSTAN=(phpstan analyse --no-progress --no-ansi --memory-limit 1G -c "${PHP_PHPSTAN_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PHP_PSALM=(psalm --config="${PHP_PSALM_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_POWERSHELL=(Invoke-ScriptAnalyzer -EnableExit -Settings "${POWERSHELL_LINTER_RULES}" -Path '{}')
+LINTER_COMMANDS_ARRAY_PROTOBUF=(protolint lint -config_path "${PROTOBUF_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_BLACK=(black --config "${PYTHON_BLACK_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_PYLINT=(pylint --rcfile "${PYTHON_PYLINT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_FLAKE8=(flake8 --config="${PYTHON_FLAKE8_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_ISORT=(isort --sp "${PYTHON_ISORT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_MYPY=(mypy --config-file "${PYTHON_MYPY_LINTER_RULES}" --install-types --non-interactive)
+LINTER_COMMANDS_ARRAY_PYTHON_PYINK=(pyink --config "${PYTHON_PYINK_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_PYTHON_RUFF=(ruff check --config "${PYTHON_RUFF_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_R=(R --slave -e "\"lints <- lintr::lint('{}');print(lints);errors <- purrr::keep(lints, ~ .\\\$type == 'error');quit(save = 'no', status = if (length(errors) > 0) 1 else 0)\"")
+LINTER_COMMANDS_ARRAY_RAKU=(raku)
+LINTER_COMMANDS_ARRAY_RENOVATE=(renovate-config-validator --strict)
+LINTER_COMMANDS_ARRAY_RUBY=(rubocop -c "${RUBY_LINTER_RULES}" --force-exclusion --ignore-unrecognized-cops)
+LINTER_COMMANDS_ARRAY_RUST_2015=(rustfmt --edition 2015)
+LINTER_COMMANDS_ARRAY_RUST_2018=(rustfmt --edition 2018)
+LINTER_COMMANDS_ARRAY_RUST_2021=(rustfmt --edition 2021)
+LINTER_COMMANDS_ARRAY_RUST_CLIPPY=(cargo clippy)
+if [ -n "${RUST_CLIPPY_COMMAND_OPTIONS:-}" ]; then
+  export RUST_CLIPPY_COMMAND_OPTIONS
+  if ! AddOptionsToCommand "LINTER_COMMANDS_ARRAY_RUST_CLIPPY" "${RUST_CLIPPY_COMMAND_OPTIONS}"; then
+    fatal "Error while adding options to Rust Clippy command"
+  fi
+fi
+LINTER_COMMANDS_ARRAY_SCALAFMT=(scalafmt --config "${SCALAFMT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_SHELL_SHFMT=(shfmt)
+LINTER_COMMANDS_ARRAY_SNAKEMAKE_LINT=(snakemake --lint -s)
+LINTER_COMMANDS_ARRAY_SNAKEMAKE_SNAKEFMT=(snakefmt --config "${SNAKEMAKE_SNAKEFMT_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_STATES=(asl-validator --json-path)
+LINTER_COMMANDS_ARRAY_SQLFLUFF=(sqlfluff)
+LINTER_COMMANDS_ARRAY_TEKTON=(tekton-lint)
+LINTER_COMMANDS_ARRAY_TERRAFORM_FMT=(terraform fmt)
+LINTER_COMMANDS_ARRAY_TERRAFORM_TFLINT=("TF_DATA_DIR=\"/tmp/.terraform-TERRAFORM_TFLINT-{//}\"" tflint -c "${TERRAFORM_TFLINT_LINTER_RULES}" "--filter=\"{/}\"")
+LINTER_COMMANDS_ARRAY_TERRAFORM_TERRASCAN=(terrascan scan -i terraform -t all -c "${TERRAFORM_TERRASCAN_LINTER_RULES}" -f)
+LINTER_COMMANDS_ARRAY_TERRAGRUNT=(terragrunt hclfmt --terragrunt-check --terragrunt-log-level error --terragrunt-hclfmt-file)
+LINTER_COMMANDS_ARRAY_TSX=(eslint -c "${TSX_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_TYPESCRIPT_ES=(eslint -c "${TYPESCRIPT_ES_LINTER_RULES}")
+LINTER_COMMANDS_ARRAY_TYPESCRIPT_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_TYPESCRIPT_STANDARD=(ts-standard --parser @typescript-eslint/parser --plugin @typescript-eslint/eslint-plugin --project "${TYPESCRIPT_STANDARD_TSCONFIG_FILE}")
+LINTER_COMMANDS_ARRAY_VUE_PRETTIER=("${PRETTIER_COMMAND[@]}")
+LINTER_COMMANDS_ARRAY_XML=(xmllint)
+LINTER_COMMANDS_ARRAY_YAML=(yamllint -c "${YAML_LINTER_RULES}" -f parsable)
+if [ "${YAML_ERROR_ON_WARNING}" == 'true' ]; then
+  LINTER_COMMANDS_ARRAY_YAML+=(--strict)
+fi
+LINTER_COMMANDS_ARRAY_YAML_PRETTIER=("${PRETTIER_COMMAND[@]}")

--- a/lib/functions/worker.sh
+++ b/lib/functions/worker.sh
@@ -149,7 +149,9 @@ function LintCodebase() {
   debug "PARALLEL_COMMAND for ${FILE_TYPE} after updating the working directory: ${PARALLEL_COMMAND[*]}"
 
   # shellcheck source=/dev/null
-  source /action/lib/functions/linterCommands.sh
+  if ! source /action/lib/functions/linterCommands.sh; then
+    fatal "Error while sourcing linter commands"
+  fi
   # Dynamically add arguments and commands to each linter command as needed
   if ! InitFixModeOptionsAndCommands "${FILE_TYPE}"; then
     fatal "Error while inizializing fix mode and check only options and commands before running linter for ${FILE_TYPE}"

--- a/test/run-super-linter-tests.sh
+++ b/test/run-super-linter-tests.sh
@@ -191,6 +191,12 @@ run_test_case_gitleaks_custom_log_level() {
   COMMAND_TO_RUN+=(--env GITLEAKS_LOG_LEVEL="warn")
 }
 
+run_test_case_linter_command_options() {
+  run_test_cases_expect_success
+  # Pick one arbitrary linter to pass options to
+  COMMAND_TO_RUN+=(--env KUBERNETES_KUBECONFORM_OPTIONS="-ignore-missing-schemas -schema-location 'https://raw.githubusercontent.com/datreeio/CRDs-catalog/main/{{.Group}}/{{.ResourceKind}}_{{.ResourceAPIVersion}}.json' --ignore-filename-pattern '.*tpl\.yaml'")
+}
+
 run_test_case_fix_mode() {
   VERIFY_FIX_MODE="true"
 

